subsys/bluetooth/host/gatt.c

```diff
@@ -2489,48 +2489,37 @@ static void gatt_sub_remove(struct bt_conn *conn, struct gatt_sub *sub,
 }
 
 #if defined(CONFIG_BT_GATT_CLIENT)
-static struct gatt_sub *gatt_sub_find_free(struct bt_conn *conn,
```

Comment:
Not sure if that is a real bug, the free_sub branch does not return so it will continue to end of the loop and only if no sub is found the free_sub is utilized.

Reply:
You are right. I have rephrased the commit message to be just a refactoring, since I need to add subscription without having an active connection.

Reply:
Fair enough.